it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManager.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import static com.google.common.hash.Hashing.goodFastHash;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.InstanceConfigId;
+import com.google.cloud.spanner.InstanceId;
+import com.google.cloud.spanner.InstanceInfo;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.CreateDatabaseException;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.CreateInstanceException;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.UpdateDatabaseException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.CharMatcher;
+import com.google.common.collect.ImmutableList;
+import com.google.common.hash.HashFunction;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default class for implementation of {@link SpannerResourceManager} interface.
+ *
+ * <p>There are one instance, one database, and multiple tables per manager object. The tables are
+ * under the database, and the database is under the instance. Instance creation is handled by
+ * {@link Builder#build()} using {@link DefaultSpannerResourceManager#createInstance(String,
+ * String)}. Database creation is triggered by the {@link
+ * DefaultSpannerResourceManager#createTable(String)} method, if there is no existing database.
+ *
+ * <p>The instance and database ids are formed using testId. The database id string will be
+ * {testId}, with some extra formatting. The instance id will be "{testId}-{ISO8601 time,
+ * microsecond precision}", with additional formatting.
+ *
+ * <p>Note: If testId is more than 30 characters, a new testId will be formed for naming: {first 22
```

Comment:
This paragraph can be added to the last one.

You should also make a paragraph saying that this class is thread safe. That's the guarantee we're offering, but we normally assume classes aren't thread safe.